internal/services/v1/expreflection.go

```diff
@@ -0,0 +1,773 @@
+package v1
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"golang.org/x/exp/maps"
+
+	"github.com/authzed/spicedb/pkg/caveats"
+	caveattypes "github.com/authzed/spicedb/pkg/caveats/types"
+	"github.com/authzed/spicedb/pkg/datastore"
+	"github.com/authzed/spicedb/pkg/diff"
+	caveatdiff "github.com/authzed/spicedb/pkg/diff/caveats"
+	nsdiff "github.com/authzed/spicedb/pkg/diff/namespace"
+	"github.com/authzed/spicedb/pkg/namespace"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+	iv1 "github.com/authzed/spicedb/pkg/proto/impl/v1"
+	"github.com/authzed/spicedb/pkg/spiceerrors"
+	"github.com/authzed/spicedb/pkg/tuple"
+	"github.com/authzed/spicedb/pkg/zedtoken"
+)
+
+type schemaFilters struct {
+	filters []compiledFilter
+}
+
+type compiledFilter struct {
+	defNameMatch    *regexp.Regexp
+	caveatNameMatch *regexp.Regexp
+	relNameMatch    *regexp.Regexp
+	permNameMatch   *regexp.Regexp
+}
+
+func newSchemaFilters(filters []*v1.ExpSchemaFilter) (*schemaFilters, error) {
```

Comment:
no tests exercising the error paths

Reply:
The schema diff unit tests exercise this code path